src/vars.c

```diff
@@ -2465,6 +2465,8 @@ void            PrintIsbComObjExpr (
 
 Obj FuncGetCurrentLVars( Obj self )
 {
+  // Need to promote to High Vars, else bag will be freed when function exits
+  MakeHighVars(STATE(CurrLVars));
```

Comment:
At first I thought this was a very nasty change but the more I think about it, the more I am convinced we should add it, regardless of the rest of this PR, as otherwise it's very easy to write buggy code by using `GetCurrentLVars()` -- that said, I also don't like that this is called `GetCurrentLVars`, I'd prefer `GET_CURRENT_LVARS` or so, to make it clear that this is a very deep internal function...

Reply:
Unfortunately I think this function has already escaped into usage with it's current name (certainly I've used it in a few places), and I assumed as a "not all capitals" function it was acceptable to use it.

Reply:
It’s an undocumented function, and for a super low level use. No package uses it - and even if, I‘d still have no qualms renaming it...